### PR TITLE
Implement the incumbent global.

### LINKED ISSUE #10963: Implement incumbent, entry, current settings objects / globals / realms
Incumbent in particular is needed for security checks (`iframe.contentDocument`, for example).

### PR DESCRIPTION
**Pull request by Ms2ger**
Fixes #10963.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14936)

### CODE REVIEW

File: components/script/dom/bindings/settings_stack.rs [hunk 1]
```diff
@@ -64,7 +77,81 @@ impl Drop for AutoEntryScript {
...
-             .last()
+             .iter()
+             .rev()
+             .find(|entry| entry.kind == StackEntryKind::Entry)
              .map(|entry| Root::from_ref(&*entry.global))
     }).unwrap()
 }
+
+/// RAII struct that pushes and pops entries from the script settings stack.
+pub struct AutoIncumbentScript {
```
Comment (nox) [inline components/script/dom/bindings/settings_stack.rs]: Shouldn't that be `#[must_root]`?

Comment (Ms2ger) [inline components/script/dom/bindings/settings_stack.rs]: There might be a better way to write it; the important thing is that we never dereference the field, we just compare / print the address.

Comment (jdm) [inline components/script/dom/bindings/settings_stack.rs]: Store it as a usize, in that case?

**Review by jdm — DISMISSED** [review]
This implementation makes sense to me. A bit more documentation wouldn't go amiss, however!

**Comment by Ms2ger** [comment]
I've added a few step comments; let me know if that's sufficient.

**Comment by jdm** [comment]
Grumble. Rebase and we'll try to merge it again!

**Comment by Ms2ger** [comment]
@bors-servo r=jdm

Tidy failed.

**Pull request merged**